includes/Compatibility.php

```diff
@@ -0,0 +1,431 @@
+	 */
+	protected $extensions = [];
+
+	/**
+	 * Array of required files.
+	 *
+	 * @var array
+	 */
+	protected $required_files = [];
+
+	/**
+	 * Class name.
+	 *
+	 * @var string
+	 */
+	protected $class_name;
+
+
+	/**
+	 * Compatibility constructor.
+	 *
+	 * @param WP_Error $error       WP_Error object passed back.
+	 */
+	public function __construct( WP_Error $error ) {
+		$this->error = $error;
+	}
+
+	/**
+	 * Check to see if PHP version check passes.
+	 *
+	 * @return bool
+	 */
+	public function check_php_version() {
+		if ( version_compare( PHP_VERSION, $this->get_php_version(), '<' ) ) {
+			/* translators: %s: PHP version number */
+			$message = esc_html( sprintf( __( 'Web Stories requires PHP %s or higher.', 'web-stories' ), $this->get_php_version() ) );
+			$data    = [
+				'title' => $message,
+			];
+			$this->error->add( 'failed_check_php_version', $message, $data );
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see if WordPress version check passes.
+	 *
+	 * @return bool
+	 */
+	public function check_wp_version() {
+		if ( version_compare( get_bloginfo( 'version' ), $this->get_wp_version(), '<' ) ) {
+			/* translators: %s: WordPress version number */
+			$message = esc_html( sprintf( __( 'Web Stories requires WordPress %s or higher.', 'web-stories' ), $this->get_wp_version() ) );
+			$data    = [
+				'title' => $message,
+			];
+			$this->error->add( 'failed_check_wp_version', $message, $data );
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see PHP has been built with composer.
+	 *
+	 * @return bool
+	 */
+	public function check_php_built() {
+		if ( ! class_exists( $this->get_class_name() ) ) {
+			$message =
+				sprintf(
+				/* translators: %s: build commands. */
+					__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
+					'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
+				);
+			$data = [
+				'title' => esc_html__( 'Web Stories plugin could not be initialized.', 'web-stories' ),
+			];
+			$this->error->add( 'failed_build', $message, $data );
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if required files.
+	 *
+	 * @return bool
+	 */
+	public function check_required_files() {
+		$required_files = $this->get_required_files();
+		if ( $required_files ) {
+			foreach ( $required_files as $required_file ) {
+				if ( ! is_readable( $required_file ) ) {
+					$message =
+						sprintf(
+						/* translators: %s: build commands. */
+							__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
+							'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
+						);
+					$data = [
+						'title' => esc_html__( 'Web Stories plugin could not be initialized.', 'web-stories' ),
+					];
+					$this->error->add( 'failed_build', $message, $data );
```

Comment:
```suggestion
					$this->error->add( 'failed_check_required_files', $message, $data );
```

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/4930/commits/5ac0fb580a6256f214599340fe5aca35e888a5e8